integration/gemfile_lock_test.go

```diff
@@ -0,0 +1,98 @@
+package integration
+
+import (
+	"fmt"
+	"os"
+	"path/filepath"
+	"testing"
+
+	"github.com/paketo-buildpacks/occam"
+	"github.com/sclevine/spec"
+
+	. "github.com/onsi/gomega"
+	. "github.com/paketo-buildpacks/occam/matchers"
+)
+
+func testGemfileLock(t *testing.T, context spec.G, it spec.S) {
+	var (
+		Expect     = NewWithT(t).Expect
+		Eventually = NewWithT(t).Eventually
+
+		pack   occam.Pack
+		docker occam.Docker
+	)
+
+	it.Before(func() {
+		pack = occam.NewPack().WithVerbose()
+		docker = occam.NewDocker()
+	})
+
+	context("when the source code contains a Gemfile.lock", func() {
+		var (
+			image     occam.Image
+			container occam.Container
+
+			name   string
+			source string
+		)
+
+		it.Before(func() {
+			var err error
+			name, err = occam.RandomName()
+			Expect(err).NotTo(HaveOccurred())
+		})
+
+		it.After(func() {
+			Expect(docker.Container.Remove.Execute(container.ID)).To(Succeed())
+			Expect(docker.Image.Remove.Execute(image.ID)).To(Succeed())
+			Expect(docker.Volume.Remove.Execute(occam.CacheVolumeNames(name))).To(Succeed())
+			Expect(os.RemoveAll(source)).To(Succeed())
+		})
+
+		it("installs the version specified therein", func() {
+			var err error
+			source, err = occam.Source(filepath.Join("testdata", "gemfile_lock_version"))
+			Expect(err).NotTo(HaveOccurred())
+
+			var logs fmt.Stringer
+			image, logs, err = pack.WithNoColor().Build.
+				WithPullPolicy("never").
+				WithBuildpacks(
+					settings.Buildpacks.RVM.Online,
+					settings.Buildpacks.Bundler.Online,
+					settings.Buildpacks.BuildPlan.Online,
+				).
+				Execute(name, source)
+			Expect(err).ToNot(HaveOccurred(), logs.String)
+
+			container, err = docker.Container.Run.
+				WithEnv(map[string]string{"PORT": "8080"}).
+				WithPublish("8080").
+				WithPublishAll().
+				Execute(image.ID)
+			Expect(err).NotTo(HaveOccurred())
+
+			// Expected a version of `bundler` which corresponds to `gems` v.3.0.8
+			Eventually(container).Should(BeAvailable())
+			Eventually(container).Should(Serve(ContainSubstring("/layers/com.anynines.buildpacks.rvm/rvm/gems/")).OnPort(8080))
+			Eventually(container).Should(Serve(MatchRegexp(`Bundler version 1\.17\.\d+`)).OnPort(8080))
+
+			Expect(logs).To(ContainLines(
+				MatchRegexp(fmt.Sprintf(`%s \d+\.\d+\.\d+`, settings.Buildpack.Name)),
+			))
+			Expect(logs).To(ContainLines(
+				MatchRegexp(`    Installing RubyGems 3.0.8`),
+				MatchRegexp(`      Successfully built RubyGem`),
+				MatchRegexp(`      Name: bundler`),
+				MatchRegexp(`      Version: 1.17.3`),
+			))
+			Expect(logs).To(ContainLines(
+				"  Using config/puma.rb supplied by application",
+				"  Adding Puma version: '4.3.5' to Gemfile",
```

Comment:
A new version of Puma will not pass this test. Maybe use a RegEx instead?

Reply:
Supplied sample application relies on default puma version. Many of these integration tests rely on specific versions of software, set in configuration files and application files (like Gemfile). This specific test definitely will fail if somebody changes default puma version (literally changes code of sample application) and doesn't run scripts/integration.sh to be sure everything is OK. So some tests will fail if another files are changed and tests aren't run after.

I'm not good in test plans, and i prefer you to decide :)
- We can leave it as is, but every person who changes repo has to run all tests before commit;
- I change puma version to x.x.x format (current is 5.x.x), but there are many another similar checks, which test specific application in specific environment (Ruby version for example).

Reply:
> I change puma version to x.x.x format (current is 5.x.x), but there are many another similar checks, which test specific application in specific environment (Ruby version for example).

Please implement this solution so Puma can be updated to a newer version without breaking the tests

Reply:
Version check for puma changed to '\d+\.\d+\.\d+' in all tests